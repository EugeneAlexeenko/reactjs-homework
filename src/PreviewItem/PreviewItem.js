import React from 'react';
import PropTypes from 'prop-types';
import { getFullYear } from '../utils';
import './PreviewItem.scss';

const PreviewItem = props => {
  const {
    poster_path,
    release_date,
    genres,
    title
  } = props.movie;

  return (
    <div className="item-preview">
      <div className="item-preview__image-container">
        <img className='item-preview__image' src={poster_path} alt={title}/>
      </div>
      <div className="item-preview__title-container">
        <div className="item-preview__title">{title}</div>
        <div className="item-preview__release-date">{getFullYear(release_date)}</div>
      </div>
      <div className="item-preview__genres-container">
        {genres.map(genre => (
          <div key={genre} className="item-preview__genre">{genre}</div>
        ))}
      </div>
    </div>
  );
};

PreviewItem.propTypes = {
  movie: PropTypes.shape({
    poster_path: PropTypes.string,
    title: PropTypes.string,
    release_date: PropTypes.string,
    genres: PropTypes.arrayOf(PropTypes.string),
  })
};

export default PreviewItem;
