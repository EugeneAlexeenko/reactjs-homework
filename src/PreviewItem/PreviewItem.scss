.item-preview {
  width: 220px;
  margin-bottom: 20px;
  color: #767676;

  &__image-container {
    overflow: hidden;
    margin-bottom: 10px;
  }

  &__image {
    width: 100%;
    height: auto;
  }

  &__title-container {
    padding: 5px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  &__title {
    font-size: 14px;
    text-transform: uppercase;
  }

  &__release-date {
    height: 10px;
    padding: 2px;
    font-size: 9px;
    border: 1px solid #767676;
    border-radius: 2px;
  }

  &__genres-container {
    padding: 5px;
    font-size: 12px;
    color: #b0b0b0;
  }

  &__genre {

  }

}
