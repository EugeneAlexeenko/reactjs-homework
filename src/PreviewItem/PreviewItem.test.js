import React from 'react';
import { shallow } from 'enzyme';
import PreviewItem from './PreviewItem';

describe('PreviewItem', () => {

  it('Renders correctly', () => {
    const testMovie = {
      budget: 100000000,
      genres: ["Action", "Thriller", "Science Fiction"],
      id: 280,
      overview: "Nearly 10 years have passed since Sarah Connor was targeted for termination by a cyborg from the future. Now her son, John, the future leader of the resistance, is the target for a newer, more deadly terminator. Once again, the resistance has managed to send a protector back to attempt to save John and his mother Sarah.",
      poster_path: "https://image.tmdb.org/t/p/w500/2y4dmgWYRMYXdD1UyJVcn2HSd1D.jpg",
      release_date: "1991-07-03",
      revenue: 520000000,
      runtime: 137,
      tagline: "It's nothing personal.",
      title: "Terminator 2: Judgment Day",
      vote_average: 7.8,
      vote_count: 5108,
    };


    const component = shallow( <PreviewItem movie={testMovie}></PreviewItem>);

    expect(component).toMatchSnapshot();
  });
});


