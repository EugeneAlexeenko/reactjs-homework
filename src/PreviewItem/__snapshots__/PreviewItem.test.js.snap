// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewItem Renders correctly 1`] = `ShallowWrapper {}`;
